components/style/animation.rs

```diff
@@ -180,6 +367,127 @@ impl Interpolate for f64 {
     }
 }
 
+impl Interpolate for i32 {
+    #[inline]
+    fn interpolate(&self, other: &i32, time: f64) -> Option<i32> {
+        let a = *self as f64;
+        let b = *other as f64;
+        Some((a + (b - a) * time).round() as i32)
+    }
+}
+
+impl Interpolate for Visibility {
+    #[inline]
+    fn interpolate(&self, other: &Visibility, time: f64)
+                   -> Option<Visibility> {
+        match (*self, *other) {
+            (Visibility::visible, _) | (_, Visibility::visible) => {
+                if time >= 0.0 && time <= 1.0 {
+                    Some(Visibility::visible)
+                } else if time < 0.5 {
```

Comment:
This is maybe an odd way of expressing it,  I'll change it to `if time < 0.0`, which is clearer. (According to http://dev.w3.org/csswg/css-transitions/#animtype-visibility , visibility should be interpolated to the nearest endpoint if time is outside [0,1])